netbox/dcim/tables.py

```diff
@@ -643,41 +643,95 @@ class Meta(BaseTable.Meta):
 # Device components
 #
 
+class DeviceComponentDetailTable(BaseTable):
+    component_type = ''
+
+    pk = ToggleColumn()
+    cable = tables.LinkColumn()
+
+    class Meta(BaseTable.Meta):
+        order_by = ('device', 'name')
+        fields = ('pk', 'device', 'name', 'type', 'description', 'cable')
+        sequence = ('pk', 'device', 'name', 'type', 'description', 'cable')
+
+
 class ConsolePortTable(BaseTable):
 
     class Meta(BaseTable.Meta):
         model = ConsolePort
         fields = ('name', 'type')
 
 
+class ConsolePortDetailTable(DeviceComponentDetailTable):
+    component_type = 'Console Ports'
+    device = tables.LinkColumn()
+
+    class Meta(DeviceComponentDetailTable.Meta, ConsolePortTable.Meta):
+        pass
+
+
 class ConsoleServerPortTable(BaseTable):
 
     class Meta(BaseTable.Meta):
         model = ConsoleServerPort
         fields = ('name', 'description')
 
 
+class ConsoleServerPortDetailTable(DeviceComponentDetailTable):
+    component_type = 'Console Server Ports'
+    device = tables.LinkColumn()
+
+    class Meta(DeviceComponentDetailTable.Meta, ConsoleServerPortTable.Meta):
+        pass
+
+
 class PowerPortTable(BaseTable):
 
     class Meta(BaseTable.Meta):
         model = PowerPort
         fields = ('name', 'type')
 
 
+class PowerPortDetailTable(DeviceComponentDetailTable):
+    component_type = 'Power Ports'
+    device = tables.LinkColumn()
+
+    class Meta(DeviceComponentDetailTable.Meta, PowerPortTable.Meta):
+        pass
+
+
 class PowerOutletTable(BaseTable):
 
     class Meta(BaseTable.Meta):
         model = PowerOutlet
         fields = ('name', 'type', 'description')
 
 
+class PowerOutletDetailTable(DeviceComponentDetailTable):
+    component_type = 'Power Outlets'
+    device = tables.LinkColumn()
+
+    class Meta(DeviceComponentDetailTable.Meta, PowerOutletTable.Meta):
+        pass
+
+
 class InterfaceTable(BaseTable):
 
     class Meta(BaseTable.Meta):
         model = Interface
         fields = ('name', 'type', 'lag', 'enabled', 'mgmt_only', 'description')
 
 
+class InterfaceDetailTable(DeviceComponentDetailTable):
+    component_type = 'Interfaces'
+    parent = tables.LinkColumn(order_by=('device', 'virtual_machine'))
+
+    class Meta(InterfaceTable.Meta, DeviceComponentDetailTable.Meta):
```

Comment:
Is there a need to inherit from other Meta classes here, since we're setting the parameters anyway? If there is, inheritance order should be consistent with the other component tables.

Reply:
I believe we are taking the model from some and the columns from another, but for this one most likely not.

I believe I set it mostly for a "in the future case" and to maintain consistency with the other models but it looks like I goofed on the ordering.